### PR TITLE
Add github actions for tests and release

### PR DESCRIPTION
**Pull request by sudeephb**
Add actions from bootstack-charms-spec

### CODE REVIEW

**Comment by sudeephb** [comment]
I have manually added the maintainers and admin permissions for the relevant teams.

**Review by Pjack — APPROVED** [review]

**Review by rgildein — CHANGES_REQUESTED** [review]
Ci needs to be enabled first.

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# This is a template `CODEOWNERS` file for ops charms
+# This file is managed by bootstack-charms-spec and should not be modified
+# within individual charm repos. https://launchpad.net/bootstack-charms-spec
+
+# For more information about CODEOWNER, please refer to
+# https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners#example-of-a-codeowners-file
+
+# These owners will be the default owners for everything in the repo. Unless a
+# later match takes precedence, @canonical/bootstack will be requested for
```
Comment (agileshaw) [inline .github/CODEOWNERS:9]: This needs to be updated as well
```suggestion
# later match takes precedence, @canonical/soleng-reviewers will be requested for
```

**Review by agileshaw — CHANGES_REQUESTED** [review]
Admin needs to add JIRA_URL to Github secrets and set up sonar project to enable those integrations

**Review by Pjack — APPROVED** [review]

File: .github/CODEOWNERS (hunk 1 above)
Comment (jneo8) [reply .github/CODEOWNERS:9]: `JIRA_URL` is there.

**Comment by jneo8** [comment]
Please move the discussion to #15 , the checks is running there.

**Pull request closed without merging**